tests/template.py

```diff
@@ -52,12 +56,31 @@ def test_has_valid_dats(self, dataset):
             )
 
         with open(os.path.join(dataset, "DATS.json"), "rb") as f:
-            if not validate_json(json.load(f)):
+            json_obj = json.load(f)
+            if not validate_json(json_obj):
                 pytest.fail(
                     f"Dataset {dataset} doesn't contain a valid DATS.json.",
                     pytrace=False,
                 )
 
+            # For crawled dataset, some tests should not be run as there is no way to
+            # automatically populate some of the fields
+            # For datasets crawled with Zenodo: check the formats extra property only
+            # For datasets crawled with OSF: skip validation of extra properties
+            is_osf_dataset = os.path.exists(os.path.join(dataset, '.conp-osf-crawler.json'))
+            is_zenodo_dataset = os.path.exists(os.path.join(dataset, '.conp-zenodo-crawler.json'))
+            is_valid, errors = validate_formats(json_obj) if is_zenodo_dataset else validate_non_schema_required(json_obj)
+            if not is_valid:
+                if not is_osf_dataset:
```

Comment:
no need for two ifs, can be a one-line If statement with AND `if not is_valid and not is_osf_dataset:`

Reply:
this is initially what I had but it did not work at the time. I re-changed it to see if it works now. Let's see what the tests say. Thanks!